### DISCUSSION TITLE
How I can use private dir on firejail for www-data that has no Home dir

### DISCUSSION THREAD

**Discussion by albadi132** [netblue30/firejail#4984] [Q&A]
I am developing a web application that executing commands from the user.
I tested firejail on my user and it was working great, but I couldn't set up the home dir for the www-data in apache server.

from /etc/passwd  `www-data: x :33:33:www-data:/var/www:/usr/sbin/nologin`
so the home dir should be /var/www but it give me Error: invalid private directory when use --private option.

also cant use Other user home/[user]/ as home even when given the right permissions.

any help or support how i can use --private option with www-data apache user?

**Comment by rusty-snake** [comment]
> also cant use Other user home/[user]/ as home even when given the right permissions.

Did you tried with `--allusers`?

**Reply by albadi132** [reply]
--allusers dos not let me use other user home dir.

But I search more in why i can't use /var/www as my home. and i find that user www-data  does not have permissions on www path so i gave it full permissions to the path, now it can be used as  home dir for --private option. But the new problem, even when I give permissions and run firejail , I can only read from the file and cannot modify or write it give me Error `read-only file system` .

**Reply by rusty-snake** [reply]
> I can only read from the file and cannot modify or write it give me Error read-only file system .

> **writable-var**
> Mount /var directory read-write.

**Reply by ghost** [reply]
Might need `writable-var-log` too if that's where your Apache is logging, just a FYI. On a more general note I'll add that Firejail isn't specifically designed to sandbox system daemons (see https://github.com/netblue30/firejail/issues/4440). You could consider using systemd hardening options instead, there is considerable overlap between them. See our [wiki page](https://github.com/netblue30/firejail/wiki/Comparison-of-firejail-and-systemd's-hardening-options) on that topic.
